### ISSUE TITLE
Connect "sidecar" content should not use nested interactive controls

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14101]
**Describe the bug**

Nested interactive controls (e.g. a link within a button) is not valid HTML, and any interactive control nested within another will not be announced to screen reader users.

In the connect chat area, the `<Content />` component (which shows the right-hand 'sidecar' content) nests several buttons inside a `div` with a role of `button` and a `tabindex` of 0.

There are several accessibility problems with this area of the app, but this issue focuses specifically on:
- Removing nested controls so that if a screen reader user tabs through the interface, every interactive control is announced
- Making sure any interactive control in this area can be activated by keyboard Enter/Space press in addition to mouse click

**To Reproduce**

- Go to `/connect` to view a chat
- Open some sidecar content, e.g. by clicking on a username in a chat message
- Using a screen reader, try to tab to the 'X' button (you will have to tab many times, which is another problem but not in the scope of this particular issue)
- Notice the button is not announced
- Repeat with other interactive controls in the sidecar content and notice none of it is announced to a screen reader user

**Expected behavior**

- Screen readers should be able to announce any interactive items in the sidecar content
- A user should be able to activate any of the interactive items in the sidecar by keyboard as well as mouse

**Screenshots**

![screenshot of the connect interface, with the user profile sidecar open and highlighted](https://user-images.githubusercontent.com/20773163/123778879-7c473680-d8c9-11eb-8395-2f87108ee0b1.png)

**Additional context**

After an upgrade to `axe-core`, this error caused some of our frontend tests in `app/javascript/chat/content.jsx` to fail. When this issue is picked up, the fix should also include re-enabling the accessibility test in that file.

**Comment by cmgorton** [comment]
I am going to close this issue since we are moving to deprecate connect. 
[Deprecating Connect Chat](https://forem.dev/foremteam/deprecating-connect-chat-2kgi)
